### PR TITLE
gromacs: add conflict between NVSHMEM and cuFFTMp

### PR DESCRIPTION
**Pull request by mabraham**
These don't work in the same build configuration.

### CODE REVIEW

**Comment by mabraham** [comment]
Context: https://gitlab.com/gromacs/gromacs/-/merge_requests/4156

File: var/spack/repos/builtin/packages/gromacs/package.py [hunk 1]
```diff
@@ -164,6 +164,8 @@ class Gromacs(CMakePackage, CudaPackage):
         description="Enable nvshmem support for nvidia gpus",
         when="+cuda+mpi",
     )
+    conflicts("+nvshmem", when="+cufftmp", msg="The GROMACS suport for NVSHMEM does not work with the GROMACS support for cuFFTMp (even though cuFFTMp uses NVSHMEM in its implementation)")
```
Comment (junghans) [inline var/spack/repos/builtin/packages/gromacs/package.py]:
```suggestion
    conflicts("+nvshmem", when="+cufftmp", msg="The GROMACS support for NVSHMEM does not work with the GROMACS support for cuFFTMp (even though cuFFTMp uses NVSHMEM in its implementation)")
```

**Review by junghans — DISMISSED** [review]

**Comment by junghans** [comment]
@spack-bot fix style

**Comment by junghans** [comment]
@spackbot fix style

**Comment by snehring** [comment]
While you're in there, could you further restrict the nvshmem variant to be `when=@2023:+mpi+cuda`. I had intended to have that and a similar conflict as part of that PR but it got lost when I decided to do fewer things in the PR than I originally was.

**Comment by mabraham** [comment]
> While you're in there, could you further restrict the nvshmem variant to be `when=@2023:+mpi+cuda`. I had intended to have that and a similar conflict as part of that PR but it got lost when I decided to do fewer things in the PR than I originally was.

Done!

File: var/spack/repos/builtin/packages/gromacs/package.py [hunk 2]
```diff
@@ -161,9 +161,16 @@ class Gromacs(CMakePackage, CudaPackage):
     variant(
         "nvshmem",
         default=False,
-        description="Enable nvshmem support for nvidia gpus",
+        when="@2023:+mpi+cuda",
```
Comment (al42and) [inline var/spack/repos/builtin/packages/gromacs/package.py]:
```suggestion
        when="@2024:+mpi+cuda",
```

The commit introducing the initial plumbing to 2024: https://gitlab.com/gromacs/gromacs/-/commit/671cfde1aafe6bb0e5a0e2063bfac5435452fd5b

Comment (snehring) [reply var/spack/repos/builtin/packages/gromacs/package.py]: 2024 is correct, I'm going to blame mobile on this one.

**Review by junghans — DISMISSED** [review]

**Comment by junghans** [comment]
@spackbot fix style

**Comment by mabraham** [comment]
@spackbot fix style

**Review by junghans — APPROVED** [review]

**Pull request merged**